cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -647,9 +647,11 @@ filterCommonFlags flags cabalLibVersion
     flags_latest = flags
     flags_3_13_0 =
       flags_latest
-        { setupWorkingDir = NoFlag
+        { -- Cabal < 3.13 does not support the --working-dir flag.
+          setupWorkingDir = NoFlag
+        , -- Or the --keep-temp-files flag.
+          setupKeepTempFiles = NoFlag
```

Comment:
I am not sure this is accurate, some commands did support `--keep-tmp-files` before? 

What happens if you are using `repl -keep-tmp-files` with an older Cabal version? Is that broken by this patch?

Reply:
_Some_ commands did; my understanding is that the point of this patch is to extend that to all commands and ensure they behave consistently.

Reply:
See also #10209.

Reply:
I've added a test that I believe will exercise this case and run it with `--boot-cabal-lib`:

https://github.com/MercuryTechnologies/cabal/blob/8255fc1e83df0f9f5f29222b128815ce37978cc6/cabal-testsuite/PackageTests/MultiRepl/CustomSetupKeepTempFiles/cabal.test.hs

Let me know if this is sufficient, or if I missed a detail.

Here's the patch to `validate.sh` I used:

```diff
diff --git a/validate.sh b/validate.sh
index b22e033f8..40c4bfcb1 100755
--- a/validate.sh
+++ b/validate.sh
@@ -505,7 +505,7 @@ CMD="$($CABALLISTBIN cabal-install:test:integration-tests2) -j1 --hide-successes
 step_cli_suite() {
 print_header "cabal-install: cabal-testsuite"
 
-CMD="$($CABALLISTBIN cabal-testsuite:exe:cabal-tests) --builddir=$CABAL_TESTSUITE_BDIR --with-cabal=$($CABALLISTBIN cabal-install:exe:cabal) $TESTSUITEJOBS  --with-ghc=$HC --hide-successes --intree-cabal-lib=$PWD --test-tmp=$PWD/testdb $RTSOPTS"
+CMD="$($CABALLISTBIN cabal-testsuite:exe:cabal-tests) --builddir=$CABAL_TESTSUITE_BDIR --with-cabal=$($CABALLISTBIN cabal-install:exe:cabal) $TESTSUITEJOBS  --with-ghc=$HC --hide-successes --boot-cabal-lib $RTSOPTS PackageTests/MultiRepl/CustomSetupKeepTempFiles/cabal.test.hs"
 (cd cabal-testsuite && timed $CMD) || exit 1
 }
```